enterprise/backend/src/metabase_enterprise/embed/content_translation/api.clj

```diff
@@ -0,0 +1,37 @@
+(ns metabase-enterprise.embed.content-translation.api
+  "Enterprise endpoints that use [JSON web tokens](https://jwt.io/introduction/) to fetch data needed for embedded
+  content translation. See the documentation for metabase.embedding.api.embed."
+  (:require
+   [metabase.api.macros :as api.macros]
+   [metabase.content-translation.models :as ct]
+   [metabase.embedding.jwt :as embedding.jwt]
+   [metabase.premium-features.core :as premium-features]
+   [metabase.util.i18n :refer [tru deferred-tru]]
+   [metabase.util.log :as log]))
+
+(set! *warn-on-reflection* true)
+
+;;; ------------------------------------------- Content translation --------------------------------------------
+
+(api.macros/defendpoint :get "/dictionary/:token"
+  "Fetch the content translation dictionary via a JSON Web Token signed with the `embedding-secret-key`."
+  [{:keys [token]} :- [:map
+                       [:token string?]]
+   {:keys [locale]}]
+  (log/info "hit embed ct endpoint")
+  ;; this will error if bad
+  (embedding.jwt/unsign token)
+  (if locale
+    {:data (ct/get-translations locale)}
+    (throw (ex-info (str (tru "Locale is required.")) {:status-code 400}))))
+
+(defn- +require-content-translation [handler]
+  (fn [request respond raise]
```

Comment:
Maybe we could make `(embedding.jwt/unsign token)` happen in a middleware like `+require-content-translation` and then apply it to all routes in this namespace? Juuuuust in case :)